articles/tech-resources/deep-links-self-serve.md

```diff
@@ -0,0 +1,189 @@
+1. (Optional) Have your app take action based on the original URL which SparkPost returns in its "3xx redirect" HTTP response.
+
+#### Forwarding Clicks From iOS To SparkPost
+
+When an iOS email client recognizes that your recipient has tapped on a universal link, it sends your application delegate an `NSUserActivity`. To allow SparkPost to track this universal link "click" event, you must call the SparkPost click tracking service to tell it about the activity. The click tracker will respond with a "3xx" redirect to the original URL from your email which may be useful to control app behaviour.
+
+Here's a sample `application:continueUserActivity:restorationHandler` method which calls the SparkPost click tracker to register a click event and also retrieves the original URL:
+
+```objc
+- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity
+ restorationHandler:(void (^)(NSArray *))restorationHandler {
+
+    // Ignore non-universal link activities
+    if (![userActivity.activityType isEqualToString:NSUserActivityTypeBrowsingWeb] || userActivity.webpageURL == nil) {
+        return NO;
+    }
+
+    // Create a task to call the SparkPost click tracker, expecting a 3xx redirect to your original URL.
+    // On completion, the handler will receive an NSURLResponse which knows the original target URL.
+    NSURLSessionDataTask* clickTask = [[NSURLSession sharedSession] dataTaskWithURL:[componentsWithURL:userActivity.webpageURL URL] completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (response) {
+            NSURL* originalURL = [response URL];
+            if (originalURL) {
+                NSLog(@"Target URL %@", originalURL.absoluteString);
+            }
+        }
+    }];
+
+    // Kick off the task
+    [clickTask resume];
+
+    return YES;
+}
+```
+
+With a little configuration, a single additional handler in your app, and no additional hosting requirements, you can enable and track iOS universal links through your SparkPost account just as if they were any other link.
+
+#### Forwarding Clicks From Android To SparkPost
+
+When an Android email client recognizes that a link has been clicked that is registered via your `AndoidManifest.xml` file.  For examples on configuring an Androdi App Link, see the documenation linked above.  Once the link is clicked, an `intent` is sent to the registered `Activity` to handle the click.  Once inside the activity, the tracking link can be manually "clicked" by performing a HTTP GET request on the link passed in with the `intent`.
+
+Here is a sample `Activity` with the corresponding `AsyncTask` that will perform an HTTP GET to the SparkPost click tracking service after an Android App Link has been clicked:
+```java
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
```

Comment:
Should we include the class that `onCreate` belongs to to people know where to include this code? Maybe just enclose the method in an outline class declaration.